src/app/embed/adapters/feed.adapter.ts

```diff
@@ -26,18 +32,32 @@ export class FeedAdapter implements DataAdapter {
     }
   }
 
-  processFeed(feedUrl: string, episodeGuid?: string): Observable<AdapterProperties> {
-    return this.fetchFeed(feedUrl).map(body => {
+  processFeed(feedUrl: string, episodeGuid?: string, baseProperties?: AdapterProperties): Observable<AdapterProperties> {
+    return this.fetchFeed(feedUrl).flatMap(body => {
       let props = this.parseFeed(body, episodeGuid);
       Object.keys(props).filter(k => props[k] === undefined).forEach(key => delete props[key]);
-      return props;
+      if (props.isReference) {
+        let { subscribeUrl: pointerFeedUrl, subtitle: pointerFeedName, feedArtworkUrl } = props,
+          base = { pointerFeedUrl, pointerFeedName, feedArtworkUrl };
+        return this.processFeed(props.referenceFeedUrl, props.referenceFeedGuid, base);
+      }
+      props.artworkUrl = props.artworkUrl || props.feedArtworkUrl;
+      if (baseProperties) {
+        Object.keys(baseProperties).forEach(baseProp => {
+          props[baseProp] = baseProperties[baseProp];
+        });
+      }
+      return Observable.of(props);
     }).catch(err => {
       FeedAdapter.logError(err);
       return Observable.of({}); // TODO: really ignore errors?
     });
   }
 
   fetchFeed(feedUrl: string): Observable<string> {
+    if (!feedUrl) {
+      return Observable.of("");
```

Comment:
I think I put that in there because I thought it might fix the failed get for `/undefined`? still trying to track that one down. Will remove this.